warrior/WarriorCore/exec_type_driver.py

```diff
@@ -11,65 +11,349 @@
+    """
+    def __init__(self, else_action):
+        """Custom exception that used to break through the rule parsing loop"""
+        super(ElseException, self).__init__(else_action)
+        self.action = else_action
+
+def math_decision(exec_condition, exec_cond_var, operator):
+    """
+        Handle the math operator decision
+        :param:
+            exec_condition: value from data_repo to be compared
+            exec_cond_var: user provided value to be compared
+            operator: math operator in plain English
+        :return:
+            True if operator condition match
+            for repo value on left and user value on right
+            Else False
+    """
+    operator = operator.lower()
+    if operator == "ge":
+        return True if get_object_from_datarepository(exec_condition) >= exec_cond_var else False
+    elif operator == "gt":
+        return True if get_object_from_datarepository(exec_condition) > exec_cond_var else False
+    elif operator == "le":
+        return True if get_object_from_datarepository(exec_condition) <= exec_cond_var else False
+    elif operator == "lt":
+        return True if get_object_from_datarepository(exec_condition) < exec_cond_var else False
+
+    pNote("Unknown error occur when deciding value, please check condition value of the step",
+          "Error")
+    return False
+
+def logical_decision(exec_condition, exec_cond_var, operator="equal"):
+    """
+        Handle the logical decision for the value comparison
+        :param:
+            exec_condition: value from data_repo to be compared
+            exec_cond_var: user provided value to be compared
+            operator: math operator in plain English
+        :return:
+            True if condition match, else return False
+    """
+    status = True
+    result = None
+    if type(get_object_from_datarepository(exec_condition)) != type(exec_cond_var):
+        pNote("Comparing different type of value, please check the conditional value type", "ERROR")
+        status = False
+    elif operator in ["ge", "gt", "le", "lt"] and\
+         not isinstance(exec_condition, int) and not isinstance(exec_condition, float):
+        pNote("Comparing non-numerical value using numerical operator,"\
+            "please check value and operator type", "ERROR")
+        status = False
+
+    if status and operator == "eq":
+        result = True if get_object_from_datarepository(exec_condition) == exec_cond_var else False
+    elif status and operator == "ne":
+        result = True if get_object_from_datarepository(exec_condition) != exec_cond_var else False
+    elif status and operator in ["ge", "gt", "le", "lt"]:
+        result = math_decision(exec_condition, exec_cond_var, operator)
+    else:
+        pNote("Execution condition failed for expected value: {} , operator: {}, actual value: {}"\
+              .format(exec_cond_var, operator,
+                      get_object_from_datarepository(exec_condition)), "WARNING")
+
```

Comment:
'eq' and 'ne' can also be handled in math_decision instead of duplicating it here.

Reply:
please refer to L72, eq and ne can be used in other data type and math_decision is only used when data type is int/float
hence the separation